### PR TITLE
qutebrowser: 2.5.4 -> 3.0.0

### PR DESCRIPTION
**Pull request by rnhmjoj**
## Description of changes

Version bump

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by mdarocha — APPROVED** [review]

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 250171` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>qutebrowser</li>
    <li>qutebrowser.dist</li>
  </ul>
</details>

File: pkgs/applications/networking/browsers/qutebrowser/default.nix [hunk 1]
```diff
@@ -100,10 +77,6 @@ buildPythonApplication {
   dontWrapGApps = true;
   dontWrapQtApps = true;
 
-  preConfigure = lib.optionalString isQt6 ''
```
Comment (nrdxp) [inline pkgs/applications/networking/browsers/qutebrowser/default.nix:103]: Was this causing an issue? This prevents the man page from building and being included in the final package, so we should probably just add this back if not.

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -1473,6 +1473,7 @@ mapAliases ({
   quake3game = throw "'quake3game' has been renamed to/replaced by 'ioquake3'"; # Converted to throw 2022-02-22
   quaternion-git = throw "quaternion-git has been removed in favor of the stable version 'quaternion'"; # Added 2020-04-09
   quilter = throw "quilter has been removed from nixpkgs, as it was unmaintained"; # Added 2021-08-03
+  qutebrowser-qt6 = throw "'qutebrowser-qt6' has been replaced by 'qutebrowser', since the the qt5 version has been removed"; # Added 2023-08-19
```
Comment (dotlambda) [inline pkgs/top-level/aliases.nix:1476]: Why not
```suggestion
  qutebrowser-qt6 = qutebrowser; # Added 2023-08-19
```
?

Comment (rnhmjoj) [reply pkgs/top-level/aliases.nix:1476]: I followed other Qt applications that did a similar migration.

**Comment by amjoseph-nixpkgs** [comment]
This wasn't just a version bump, you broke `qutebrowser` for all platforms where `qt6` doesn't yet build.

A massively drastic change like that shouldn't be smuggled into nixpkgs in a PR which is labeled as a version bump.

Prior to this PR the expression supported both `qt5` and `qt6`.

**Pull request merged**